lib/cam.js

```diff
@@ -253,11 +253,15 @@ Cam.prototype.getSystemDateAndTime = function(callback) {
 		'</s:Envelope>'
 	}, function(err, data, xml) {
 		if (!err) {
-			var dt = linerase(data[0]['getSystemDateAndTimeResponse'][0]['systemDateAndTime'][0]['UTCDateTime'][0])
-				, time = new Date(Date.UTC(dt.date.year, dt.date.month - 1, dt.date.day, dt.time.hour, dt.time.minute, dt.time.second))
-				;
-			if (!this.timeShift) {
-				this.timeShift = time - Date.now();
+			try{
+				var dt = linerase(data[0]['getSystemDateAndTimeResponse'][0]['systemDateAndTime'][0]['UTCDateTime'][0])
+					, time = new Date(Date.UTC(dt.date.year, dt.date.month - 1, dt.date.day, dt.time.hour, dt.time.minute, dt.time.second))
+					;
+				if (!this.timeShift) {
+					this.timeShift = time - Date.now();
+				}
+			}catch(err) {
+				callback.call(this, {"err" : "err"}, null, xml);
```

Comment:
I'm not sure we should error out here, potentially it should be a warning that the system doesn't support date/time. If we error here then there is likelihood that the library will fail its init check and nothing else will work (I haven't checked)

This is also failing on CI as a syntax error so please check that too

Reply:
getSystemDateAndTime is a mandatory function, so every onvif device should support it. Therefore I think an error is the correct approach. A warning should not solve the problem, if a device responds to an onvif discovery but it's not an onvif device this will raise some exceptions in other functions.

Reply:
Some cameras cover partial ONVIF spec. Seems they have their own API for controlling their camera, and then shoe-horn code in to support ONVIF (especially China-made cameras, although their are a couple North America ones as well). The code changes above seem reasonable to me.